narwhals/_spark_like/utils.py

```diff
@@ -35,6 +37,8 @@ def native_to_narwhals_dtype(
         return dtypes.Int32()
     if isinstance(dtype, pyspark_types.ShortType):
         return dtypes.Int16()
+    if isinstance(dtype, pyspark_types.ByteType):
+        return dtypes.Int8()
```

Comment:
I remember this came up already @MarcoGorelli . I found the source: https://spark.apache.org/docs/latest/sql-ref-datatypes.html  "`ByteType` is a 1-byte signed integer (within the range of -128 to 127)."  :D also [in the  `spark.pandas` code](https://github.com/apache/spark/blob/master/python/pyspark/pandas/typedef/typehints.py#L258-L259)